test/unit/kubernetes-deploy/runner_test.rb

```diff
@@ -0,0 +1,43 @@
+require 'test_helper'
+
+class RunnerTest < Minitest::Test
+  def setup
+    @logger_stream = StringIO.new
+    @logger = Logger.new(@logger_stream)
+    KubernetesDeploy.logger = @logger
+  end
+
+  def teardown
+    @logger_stream.close
```

Comment:
If we're going to do this in the unit tests, perhaps this should be in a default `#setup` that integration tests also inherit from?